src/makernote_int.cpp

```diff
@@ -813,60 +813,52 @@ struct NikonArrayIdx {
 #define NA ((uint32_t)-1)
 
 //! Nikon binary array version lookup table
-constexpr NikonArrayIdx nikonArrayIdx[] = {
+constexpr auto nikonArrayIdx = std::array{
```

Comment:
[suggestion] I already commented this in one of your previous PRs, I do not like too much the need of repeating the type explicitly in item. Another alternative would be to be more explicit about the type used in the first line and the usage of extra braces for avoiding that:

```cpp
constexpr std::array<NikonArrayIdx,32> nikonArrayIdx {{
    // NikonSi
    {0x0091, "0208", 0, 0, 4},     // D80
    {0x0091, "0209", 0, 1, 4},     // D40
    {0x0091, "0210", 5291, 2, 4},  // D300
    {0x0091, "0210", 5303, 3, 4},  // D300, firmware version 1.10
    {0x0091, "02", 0, 4, 4},       // Other v2.* (encrypted)
    {0x0091, "01", 0, 5, NA},      // Other v1.* (not encrypted)
...
    // NikonAf
    {0x00b7, "0100", 30, 0, NA},  // These sizes have been found in tiff headers of MN
    {0x00b7, "0101", 84, 1, NA},  // tag 0xb7 in sample image metadata for each version
}};
```

Reply:
It's possible to simplify this with templating to avoid explicit size parameter and extra {}. I think this can be handled at a later time.

Reply:
I'll also note that std::pair and tuple implement constructors, so {} can be replaced with (). I've done so in some places.